core/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/list/ListTasksResponse.java

```diff
@@ -96,6 +100,41 @@ public void writeTo(StreamOutput out) throws IOException {
         return nodeTasks;
     }
 
+    public List<TaskGroup> getTaskGroups() {
+        if (groups == null) {
+            buildTaskGroups();
+        }
+        return groups;
+    }
+
+    private void buildTaskGroups() {
+        Map<TaskId, TaskGroup.Builder> taskGroups = new HashMap<>();
```

Comment:
The original list is not really sorted in any particular way since it's populated from a map. So, I didn't see any reason to preserve the original sort order.